Assets/Scripts/Controllers/Web3Controller.cs

```diff
@@ -156,7 +156,7 @@ private void OpenMetamaskConnectDialog()
 #if !UNITY_EDITOR
         if (MetamaskInterop.IsMetamaskAvailable())
         {
-            MetamaskInterop.EnableEthereum(gameObject.name, nameof(EthereumEnabled), nameof(DisplayError));
+            MetamaskInterop.EnableEthereum(nameof(Web3Controller), nameof(EthereumEnabled), nameof(DisplayError));
```

Comment:
This should refer to the game object's name in the scene.
Was this a necessary change?

Reply:
DontDestroyOnLoad makes this instance stick around during scene loads, so it destroys any "extra" instances that get initiated when we switch back to the main menu scene. The original instance (referenced by the singleton `instance` field) should still be there

Reply:
Might be better to switch the buttons to use a script (that references Web3Controller.instance) instead of unity events

Reply:
it was hard to debug because the problems only really occurred in WebGL, at some point I switched the buttons onClick events via script (for testing purposes), but switched it back to what it was again since it was working again after this change.

I will attach the onClick events using scripts then.

Reply:
Thanks